cmd/server.go

```diff
@@ -32,17 +32,20 @@ var cmdServer = &cobra.Command{
 			log.Fatal(err)
 		}
 
-		repository, err := store.NewStore(cmd.Context(), app.Config, app.Config.ConditionDefinitions, app.Logger)
+		streamBroker, err := events.NewStream(app.Config.NatsOptions)
 		if err != nil {
 			app.Logger.Fatal(err)
 		}
 
-		streamBroker, err := events.NewStream(app.Config.NatsOptions)
-		if err != nil {
+		if err = streamBroker.Open(); err != nil {
 			app.Logger.Fatal(err)
 		}
 
-		if err := streamBroker.Open(); err != nil {
+		// setup cancel context with cancel func
+		ctx, serverCancel := context.WithCancel(cmd.Context())
+
+		repository, err := store.NewStore(ctx, app.Config, app.Config.ConditionDefinitions, app.Logger, streamBroker)
```

Comment:
We require two separate repository instances available here - one for the Condition and the other for HSS server data?
That would mean `store.NewStore` has to be invoked once for each of them,
```
conditionStore, err := store.NewStore(ctx, app.Config, app.Config.ConditionDefinitions, app.Logger, streamBroker)
...
serverStore, err := store.NewStore(ctx, app.Config, app.Config.ConditionDefinitions, app.Logger, nil)
```

And we'd want to rename and create `Config.StoreKind` as two separate fields in the configuration,

https://github.com/metal-toolbox/conditionorc/blob/6d0c0e86b8508e4ba82b5301137bfe3f197e4ec2/internal/app/config.go#L39
```
StoreKindConditions model.StoreKind   
StoreKindServers model.StoreKind
```

Reply:
We don’t need 2 instances of `Store`. We either get one specialized to NATS or Serverservice. Both incorporate a client for HSS (for now), but I intentionally bury that implementation detail. The HSS client is only for server lookup.

Reply:
Ignore my comment here, I was missing the fact that the NATs store wrapped the HSS client in there.